src/lockfile/types.ts

```diff
@@ -15,3 +15,29 @@ export type ParsedLock = {
 export interface PnpmLockFile {
   packages: { [name: string]: any };
 }
+
+export interface NpmWorspacesInfo {
+  version: "string";
+  workspaces: Record<"packages", string[]>;
+}
+
+export interface NpmSynlinkInfo {
+  resolved: string; // Where the package is  reslved from.
+  link: boolean; // A flag to indicate that this is a symbolic link.
+  integrity?: 'sha512' | 'sha1';
+  dev?: boolean;
+  optional?: boolean;
+  devOptional?: boolean;
+  dependendcies?: { [key in string]: LockDependency; }
```

Comment:
This is misspelled throughout the PR. Also it should be `key: string`. 
```suggestion
  dependencies?: { [key: string]: LockDependency; }
```

Reply:
`key in string` is the original syntax used in the codebase, so I used it in the code that I added. I could replace it in the rest of the code, but that would be touching code that is out of scope of this PR…